signer/src/utxo.rs

```diff
@@ -370,6 +413,62 @@ impl<'a> UnsignedTransaction<'a> {
         }
     }
 
+    /// Constructs the set of digests that need to be signed before broadcasting
+    /// the transaction.
+    ///
+    /// # Notes
+    ///
+    /// This function uses the fact certain invariants about this struct are
+    /// upheld. They are
+    /// 1. The first input to the Transaction in the `tx` field is the signers'
+    ///    UTXO.
+    /// 2. The other inputs to the Transaction in the `tx` field are ordered
+    ///    the same order as DepositRequests in the `requests` field.
```

Comment:
Nice that you outline this. I wish there was a simple way to enforce this invariant in the type directly, but it does not seem trivial.

Reply:
I thought about it, and yeah, it's not obvious how to do so.